RIOT-OS/RIOT
# boards: Added standalone ATmega328p

### Contribution description

The PR adds the `atmega328p` "board", which allows to use the ATmega328P without any external circuitry except a stable power supply. An ICSP programmer is required to flash the board and an TTL adapter to get the serial terminal. This "board" will be much less convenient to use than an Arduino UNO or an Arduino Nano (which contain this exact MCU), but it can be very convenient as a low-power companion MCU. E.g. the ATmega328p can be used to read out a bunch of slow sensors while the main (and more power demanding) MCU is in lowest power mode. The main MCU could periodically wake up to collect and process the accumulated data from the ATmegea328p.

![ATmega328p DIP package on a breadboard](https://github.com/maribu/images/raw/master/ATmega328p.jpg)

### Testing procedure

- Check the provided documentation
- Using an ICSP try if `make BOARD=atmega328p flash` for `examples/hello-world` works. Refer to the doc of this PR when not using `usbtiny` as programmer
- Using an TTL adapter check if you get the output. (Keep in mind that this is not an Arduino that conveniently reboots upon `make term`...)

As this shares most code with the Arduino UNO, no huge troubles are expected.

### Issues/PRs references

None
